filters/flowid/filter.go

```diff
@@ -0,0 +1,64 @@
+package flowid
+
+import (
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/skipper"
+)
+
+const (
+	filterName       = "FlowId"
+	flowIdHeaderName = "X-Flow-Id"
+)
+
+type flowId struct {
+	id            string
+	reuseExisting bool
+	flowIdLength  uint8
+}
+
+func New(id string, allowOverride bool, len uint8) skipper.Filter {
+	return &flowId{id, allowOverride, len}
+}
+
+func (this *flowId) Id() string { return this.id }
+
+func (this *flowId) Name() string { return filterName }
+
+func (this *flowId) Request(fc skipper.FilterContext) {
+	r := fc.Request()
+	var flowId string
+
+	if this.reuseExisting {
+		flowId = r.Header.Get(flowIdHeaderName)
+		if isValid(flowId) {
+			return
+		}
+	}
+
+	flowId, err := newFlowId(this.flowIdLength)
+	if err == nil {
+		fc.Request().Header.Set(flowIdHeaderName, flowId)
+	}
+}
+
+func (this *flowId) Response(skipper.FilterContext) {}
+
+func (this *flowId) MakeFilter(id string, fc skipper.FilterConfig) (skipper.Filter, error) {
+	var reuseExisting bool
+	if len(fc) > 0 {
+		if r, ok := fc[0].(bool); ok {
+			reuseExisting = r
+		} else {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+	var flowIdLength uint8 = defaultLen
```

Comment:
Right, it's used only here but it is in the hashing file. Will move the const here